### PR TITLE
1st batch #90

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
some docs

### CODE REVIEW

File: gui/doc/chart.csv [hunk 1]
```diff
@@ -1,13 +1,13 @@
 ﻿Name,Type,Default,Doc
 *value,any,,The data object bound to this chart control.
 title,str,None,The title of this chart control.
-width,str|int|float,100vw,"The width, in CSS units, of this element."
-height,str|int|float,100vw,"The height, in CSS units, of this element."
+width,str|int|float,100%,"The width, in CSS units, of this element."
+height,str|int|float,100vh,"The height, in CSS units, of this element."
```
Comment (FabienLelaquais) [inline gui/doc/chart.csv]: vh, really?

File: gui/doc/chart.csv [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 selected,dynamic(list[int]|str),None,List of the selected point indices.
 layout,"dict[str, any]",None,"The <i>plotly.js</i> compatible <a href=""https://plotly.com/javascript/reference/layout/"">layout object</a>."
 plot_config,"dict[str, any]",None,"The <i>plotly.js</i> compatible <a href=""https://plotly.com/javascript/configuration-options/""> configugration options object</a>."
 options,"indexed(dict[str, any])",None,"The <i>plotly.js</i> compatible <a href=""https://plotly.com/javascript/reference/"">data object where dynamic data will be overridden.</a>."
-render,bool,True,"If <i>True</i>, this chart is visible on the page.
+render,bool,True,"If <i>True</i>, this chart is visible on the page."
```
Comment (FabienLelaquais) [inline gui/doc/chart.csv]: The (new) convention is NOT to user Python keywords in italics...
Therefore "If True..."

File: gui/doc/chart.md [hunk 3]
```diff
@@ -17,20 +17,20 @@ no index) which is overridden by any specified indexed property.
 ### Simple example
 
 If you want to create a chart that represents a dataframe stored in the Python
-variable _data_, where column "Col1" you can use the following content:
+variable _data_ (first and second columns will be used for x and y values), you can use the following content:
```
Comment (FabienLelaquais) [inline gui/doc/chart.md]: There I would <i> the x and y letters, to stress a little.

File: gui/doc/chart.md [hunk 4]
```diff
@@ -82,5 +108,43 @@ are selected in a variable.
...
+        <|{data}|chart|on_range_change=range_change_function_name|...|>
+        ```
+
+    === "HTML"
+
+        ```html
+        <taipy:chart on_range_change="range_change_function_name" ...>{data}</taipy:chart>
+        ```
+
+### sampling data
```
Comment (FabienLelaquais) [inline gui/doc/chart.md]: Capital 's'

File: gui/doc/chart.md [hunk 5]
```diff
@@ -82,5 +108,43 @@ are selected in a variable.
...
+
+By specifying _limit_rows_ as True, charts will adapt the number of points to the viewable area.
+Depending on the algorithm, Some specific points mightnot be shown on the chart.
```
Comment (FabienLelaquais) [inline gui/doc/chart.md]: "mightnot" -> "might not"

File: gui/doc/dialog.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 A modal dialog.
 
-TODO: Complete
+Dialog allows to show some content over the current page. it is closed automatically on Cancel, Validation or click on the background.
```
Comment (FabienLelaquais) [inline gui/doc/dialog.md]: " it is closed automatically on Cancel, Validation or click on the background."
->
"The dialog form is closed when the user presses the Cancel or Validate buttons, or clicks outside the area of the dialog (triggering a Cancel action)."

File: gui/doc/dialog.md [hunk 7]
```diff
@@ -42,9 +42,73 @@ and what actions (callback functions) are triggered when buttons are pressed:
...
+         on_validate="validate_action"
          cancel_label="Cancel"
-         cancel_action="cancel_action">{show_dialog}</taipy:dialog>
+         on_cancel="cancel_action">{show_dialog}</taipy:dialog>
         ```
 
-<|dialog|title=Dialog Title|open={value}|page_id=page1|validate_action=validate_action|cancel_action=cancel_action||validate_action_text=Validate|cancel_action_text=Cancel|>
+### Dialog as block
+
+Dialog content can be specified as inside the dialog block
```
Comment (FabienLelaquais) [inline gui/doc/dialog.md]: "Dialog content"
->
"The content of the dialog"

"can be specified as inside the dialog block"
->
"can be define directory inside the dialog block"

File: gui/doc/dialog.md [hunk 8]
```diff
@@ -42,9 +42,73 @@ and what actions (callback functions) are triggered when buttons are pressed:
...
+
+!!! example "Page content"
+
+    === "Markdown"
+
+        ```
+        <|{show_dialog}|dialog|
+            ...
+            Some text
```
Comment (FabienLelaquais) [inline gui/doc/dialog.md]: ... some... Content ?

File: gui/doc/dialog.md [hunk 9]
```diff
@@ -42,9 +42,73 @@ and what actions (callback functions) are triggered when buttons are pressed:
...
+            ...
+            <taipy:text>{some content}</taipy:text>
+            ...
+        <taipy:dialog.end></taipy:dialog.end>
+        ```
+
+### Dialog with page
+
+Dialog content can be specified as an existing page route through the _page_ property.
```
Comment (FabienLelaquais) [inline gui/doc/dialog.md]: "Dialog content"
->
"The content of the dialog"

"existing page route through the _page_ property"
-> 
"existing page name using the _page_ property"

File: gui/doc/dialog.md [hunk 10]
```diff
@@ -42,9 +42,73 @@ and what actions (callback functions) are triggered when buttons are pressed:
...
+
+!!! example "Page content"
+
+    === "Markdown"
+
+        ```
+        <|{show_dialog}|dialog|page=route|>
```
Comment (FabienLelaquais) [inline gui/doc/dialog.md]: route -> page_name

File: gui/doc/dialog.md [hunk 11]
```diff
@@ -42,9 +42,73 @@ and what actions (callback functions) are triggered when buttons are pressed:
...
+  
+    === "HTML"
+
+        ```html
+        <taipy:dialog page="route">{show_dialog}</taipy:dialog>
+        ```
+
+### Dialog with partial
+
+Dialog content can be specified as a _Partial_ instance through the _partial_ property.
```
Comment (FabienLelaquais) [inline gui/doc/dialog.md]: "Dialog content"
->
"The content of the dialog"

through -> using?

File: gui/doc/expandable.md [hunk 12]
```diff
@@ -1,13 +1,52 @@
 Displays its child elements in a collapsable area.
```
Comment (FabienLelaquais) [inline gui/doc/expandable.md:1]: Please skip a line after the first one, so we can retrieve the short description.

File: gui/doc/expandable.md [hunk 13]
```diff
@@ -1,13 +1,52 @@
...
-<code><|title|expandable|<br>...<br>|></code>
+### Defining a title and managing expanded state
 
-### Advanced
-<code><|{value}|expandable|expanded={False}|<br>...<br>|></code>
-<br>or with properties<br>
-<code><|expandable|properties={properties}|<br>...<br>|></code>
-<br>or with closing tag<br>
-<code><|expandable.start|...|><br>...<br><|expandable.end|></code>
+The default property _title_ defines the title shown when closed.
```
Comment (FabienLelaquais) [inline gui/doc/expandable.md]: when... the visual element is collapsed.

File: gui/doc/expandable.md [hunk 14]
```diff
@@ -1,13 +1,52 @@
...
+        <|Title|expandable|expand={expand}|>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:expandable expand="{expand}">Title</taipy:expandable>
+        ```
+
+### Exapandable as block
```
Comment (FabienLelaquais) [inline gui/doc/expandable.md]: "Exapandable as block"
->
"Content as a block"

File: gui/doc/expandable.md [hunk 15]
```diff
@@ -1,13 +1,52 @@
...
+
+Expandable content is specified inside the expandable block
```
Comment (FabienLelaquais) [inline gui/doc/expandable.md]: "Expandable content is specified inside the expandable block"
->
"The content of `expandable` can be specified as the block content."  [NOTE THE FINAL '.']

File: gui/doc/expandable.md [hunk 16]
```diff
@@ -1,13 +1,52 @@
...
+
+!!! example "Page content"
+
+    === "Markdown"
+
+        ```
+        <|Title|expandable|
+            ...
+            Some text
```
Comment (FabienLelaquais) [inline gui/doc/expandable.md]: Not really useful is it?

File: gui/doc/expandable.md [hunk 17]
```diff
@@ -1,13 +1,52 @@
...
+            ...
+        |>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:expandable.start>Title</taipy:expandable.start>
+            ...
+            Some text
```
Comment (FabienLelaquais) [inline gui/doc/expandable.md]: ?

File: gui/doc/file_download.md [hunk 18]
```diff
@@ -3,3 +3,77 @@ A control that allows to download a file.
 The download can be triggered when clicking on a button, or can be performed automatically.
```
Comment (FabienLelaquais) [inline gui/doc/file_download.md]: Short doc: "Allows to download a file content."

Comment (FabienLelaquais) [inline gui/doc/file_download.md]: No '.'

File: gui/doc/file_download.md [hunk 19]
```diff
@@ -3,3 +3,77 @@ A control that allows to download a file.
...
 
 ## Usage
+
+### default behavior
```
Comment (FabienLelaquais) [inline gui/doc/file_download.md]: Default

File: gui/doc/file_download.md [hunk 20]
```diff
@@ -3,3 +3,77 @@ A control that allows to download a file.
...
+        <|{content}|file_download|>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:file_download>{content}</taipy:file_download>
+        ```
+
+### standard configuration
```
Comment (FabienLelaquais) [inline gui/doc/file_download.md]: Standard

File: gui/doc/file_download.md [hunk 21]
```diff
@@ -3,3 +3,77 @@ A control that allows to download a file.
...
+
+A specific _label_ can be shown besides the standard icon. 
+The function name provided as _on_action_ is called when the user initiate the download.
```
Comment (FabienLelaquais) [inline gui/doc/file_download.md]: "initiates"

File: gui/doc/file_download.md [hunk 22]
```diff
@@ -3,3 +3,77 @@ A control that allows to download a file.
...
+        <|{content}|file_download|label=Download File|on_action=function_name|name=filename|>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:file_download on_action="function_name"  name="filename">{content}</taipy:file_download>
+        ```
+
+### file preview in the browser
```
Comment (FabienLelaquais) [inline gui/doc/file_download.md]: "Preview file in the browser"

File: gui/doc/file_download.md [hunk 23]
```diff
@@ -3,3 +3,77 @@ A control that allows to download a file.
...
+
+File content can be visualized in the browser (if supported and in another tab) by specifying _bypass_preview_ as False.
```
Comment (FabienLelaquais) [inline gui/doc/file_download.md]: "The file content can..."

File: gui/doc/file_download.md [hunk 24]
```diff
@@ -3,3 +3,77 @@ A control that allows to download a file.
...
+        <|{content}|file_download|bypass_preview=False|>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:file_download bypass_preview="False">{content}</taipy:file_download>
+        ```
+
+### automatic download
```
Comment (FabienLelaquais) [inline gui/doc/file_download.md]: Automatic

File: gui/doc/file_download.md [hunk 25]
```diff
@@ -3,3 +3,77 @@ A control that allows to download a file.
...
+
+File content can be downloaded automatically (when the page shows or when the content is set).
```
Comment (FabienLelaquais) [inline gui/doc/file_download.md]: The file content.

Is is an 'or' or an 'and'?

File: gui/doc/file_selector.md [hunk 26]
```diff
@@ -3,3 +3,60 @@ A control that allows to upload files.
 The upload can be triggered by pressing a button, or drag-and-dropping a file on top of the control.
```
Comment (FabienLelaquais) [inline gui/doc/file_selector.md:3]: Short doc:

"Allows to upload a file content."

File: gui/doc/file_selector.md [hunk 27]
```diff
@@ -3,3 +3,60 @@ A control that allows to upload files.
...
 
 ## Usage
+
+### default behavior
```
Comment (FabienLelaquais) [inline gui/doc/file_selector.md]: Default

File: gui/doc/file_selector.md [hunk 28]
```diff
@@ -3,3 +3,60 @@ A control that allows to upload files.
...
+
+The variable specified in _content_ is populated by a local filename when transfer is finished.
```
Comment (FabienLelaquais) [inline gui/doc/file_selector.md]: "when the transfer"

File: gui/doc/file_selector.md [hunk 29]
```diff
@@ -3,3 +3,60 @@ A control that allows to upload files.
...
+        <|{content}|file_selector|>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:file_selector>{content}</taipy:file_selector>
+        ```
+
+### standard configuration
```
Comment (FabienLelaquais) [inline gui/doc/file_selector.md]: Standard

File: gui/doc/file_selector.md [hunk 30]
```diff
@@ -3,3 +3,60 @@ A control that allows to upload files.
...
+
+A specific _label_ can be shown besides the standard icon. 
+The function name provided as _on_action_ is called when the transfer is finished.
+The _extensions_ property is a list of extensions that is used to filter the file selction box but won't be enforced ie the suer can select and upload any file.
```
Comment (FabienLelaquais) [inline gui/doc/file_selector.md]: -> "The _extensions_ property can be used as a list of file name extensions that is used to filter the file selection box. This filter is not enforced: the user can select and upload any file."

File: gui/doc/file_selector.md [hunk 31]
```diff
@@ -3,3 +3,60 @@ A control that allows to upload files.
...
+        <|{content}|file_selector|label=Download File|on_action=function_name|extensions=.csv,.xlsx|drop_message=Drop Message|>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:file_selector on_action="function_name" extensions=".csv,.xlsx" drop_message="Drop Message">{content}</taipy:file_selector>
+        ```
+
+### multiple files upload
```
Comment (FabienLelaquais) [inline gui/doc/file_selector.md]: Multiple

File: gui/doc/file_selector.md [hunk 32]
```diff
@@ -3,3 +3,60 @@ A control that allows to upload files.
...
+
+The user can transfer multiple files at once by specifying the _multiple_ property as True.
```
Comment (FabienLelaquais) [inline gui/doc/file_selector.md]: "specifying the _multiple_ property as True"
->
"setting the _multiple_ property to True"

File: gui/doc/image.md [hunk 33]
```diff
@@ -4,4 +4,39 @@ You can indicate a function to be called when the user clicks on the image.
 
 ## Usage
 
-TODO
+### default behavior
```
Comment (FabienLelaquais) [inline gui/doc/image.md]: Default

File: gui/doc/image.md [hunk 34]
```diff
@@ -4,4 +4,39 @@ You can indicate a function to be called when the user clicks on the image.
...
+
+Shows an image specified as a local file path or as raw content.
+
```
Comment (FabienLelaquais) [inline gui/doc/image.md:13]: A word on how 'raw content' is used?

File: gui/doc/image.md [hunk 35]
```diff
@@ -4,4 +4,39 @@ You can indicate a function to be called when the user clicks on the image.
...
+        <|{content}|image|>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:image>{content}</taipy:image>
+        ```
+
+### standard configuration
```
Comment (FabienLelaquais) [inline gui/doc/image.md]: Call a function on click

File: gui/doc/indicator.md [hunk 36]
```diff
@@ -4,3 +4,58 @@ The _min_ value can be greater than the _max_ value.<br/>
 The value will be maintained between min and max.
 
 ## Usage
+
+### default behavior
```
Comment (FabienLelaquais) [inline gui/doc/indicator.md]: Minimal usage

File: gui/doc/indicator.md [hunk 37]
```diff
@@ -4,3 +4,58 @@ The _min_ value can be greater than the _max_ value.<br/>
...
+
+Shows a message at a specified position between min and max.
+
+!!! example "Page content"
+
+    === "Markdown"
+
+        ```
+        <|message|indicator|value={val}|min=0|max=100|>
```
Comment (FabienLelaquais) [inline gui/doc/indicator.md:17]: message?
Yeah why not

File: gui/doc/indicator.md [hunk 38]
```diff
@@ -4,3 +4,58 @@ The _min_ value can be greater than the _max_ value.<br/>
...
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:indicator value="{val}" min="0" max ="100">message</taipy:indicator>
+        ```
+
+### formatting the message
```
Comment (FabienLelaquais) [inline gui/doc/indicator.md]: Formatting

File: gui/doc/indicator.md [hunk 39]
```diff
@@ -4,3 +4,58 @@ The _min_ value can be greater than the _max_ value.<br/>
...
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:indicator format="%.2f" value="10">{50}</taipy:indicator>
+        ```
+
+
+### vertical orientation
```
Comment (FabienLelaquais) [inline gui/doc/indicator.md]: Vertical indicators

File: gui/doc/indicator.md [hunk 40]
```diff
@@ -4,3 +4,58 @@ The _min_ value can be greater than the _max_ value.<br/>
...
+
+The _orientation_ of the indicator can be specified as vertical.
```
Comment (FabienLelaquais) [inline gui/doc/indicator.md]: "can be specified as vertical."
->
"can be set to "vertical" (or "v") to create a vertical indicator."

File: gui/src/components/Taipy/Chart.tsx [hunk 41]
```diff
@@ -111,7 +111,7 @@ const Chart = (props: ChartProp) => {
         title = "",
         className,
         width = "100%",
-        height = "100%",
+        height = "100vh",
```
Comment (FabienLelaquais) [inline gui/src/components/Taipy/Chart.tsx]: vh ?

**Review by FabienLelaquais — APPROVED** [review]
Amazing job!!

File: gui/doc/chart.csv (hunk 1 above)
Comment (FredLL-Avaiga) [reply gui/doc/chart.csv]: 100% doesn't work for height usually

File: gui/doc/chart.csv (hunk 2 above)
Comment (FredLL-Avaiga) [reply gui/doc/chart.csv]: will do

File: gui/doc/chart.md (hunk 3 above)
Comment (FredLL-Avaiga) [reply gui/doc/chart.md]: you mean &lt;i> ?

File: gui/doc/expandable.md (hunk 16 above)
Comment (FredLL-Avaiga) [reply gui/doc/expandable.md]: ?

File: gui/doc/expandable.md (hunk 17 above)
Comment (FredLL-Avaiga) [reply gui/doc/expandable.md]: ?

File: gui/doc/image.md (hunk 34 above)
Comment (FredLL-Avaiga) [reply gui/doc/image.md:13]: ok

File: gui/src/components/Taipy/Chart.tsx (hunk 41 above)
Comment (FredLL-Avaiga) [reply gui/src/components/Taipy/Chart.tsx]: yes ;-)

File: gui/doc/chart.md (hunk 3 above)
Comment (FredLL-Avaiga) [reply gui/doc/chart.md]: or _x

File: gui/doc/chart.csv (hunk 1 above)
Comment (FabienLelaquais) [reply gui/doc/chart.csv]: Ok... why not empty then?

File: gui/doc/chart.md (hunk 3 above)
Comment (FabienLelaquais) [reply gui/doc/chart.md]: Sorry. That would be
"first and second columns will be used for _x_ and _y_ values"

File: gui/src/components/Taipy/Chart.tsx (hunk 41 above)
Comment (FabienLelaquais) [reply gui/src/components/Taipy/Chart.tsx]: Hmmm

File: gui/doc/chart.csv (hunk 1 above)
Comment (FredLL-Avaiga) [reply gui/doc/chart.csv]: that's a plan

**Pull request merged**